packages/overlay/src/overlay-types.ts

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { ThemeData } from '@spectrum-web-components/theme';
+import { Placement as PopperPlacement } from './popper';
+
+export type TriggerInteractions = 'click' | 'hover' | 'custom';
+
+export interface OverlayOpenDetail {
+    content: HTMLElement;
+    contentTip?: HTMLElement;
+    delayed: boolean;
+    offset: number;
+    placement?: Placement;
+    trigger: HTMLElement;
+    interaction: TriggerInteractions;
+    theme: ThemeData;
+}
+
+/**
+ * Used, via an event, to query details about how an element should be shown in
+ * an overlay
+ */
+export interface OverlayDisplayQueryDetail {
+    overlayRootName?: string;
+    overlayRootElement?: HTMLElement;
+    overlayContentTipElement?: HTMLElement;
+}
+
+export type SpectrumPlacement = 'top' | 'left' | 'right' | 'bottom';
+export type Placement = PopperPlacement | 'none';
+
+export function spectrumPlacement(
+    placement: Placement
+): SpectrumPlacement | undefined {
+    if (!placement || placement === 'none' || /^auto/.test(placement)) return;
+    const match = /^([^-]+)/.exec(placement);
+    if (!match) return;
+    return match[1] as SpectrumPlacement;
+}
```

Comment:
Can you explain a bit the purpose of this function? Seems quite complicated. Is this just to normalize spectrum placement strings with popper strings?

Reply:
This is something else we might be able to partner with Spectrum CSS on overtime. If those selectors were `computed-placement^="top"` it seems like we'd be able to avoid this?

Reply:
Basically, popper supports more placement options than Spectrum does. Our API can support all of the popper options, but we have to map the popper options `top-start`, `top-end` and `top` to the `top` option that Spectrum supports.

I hemmed and hawed a little about this function. It would be more code, but the purpose would be more obvious if I just made it a big mapping.

Westbrook is right that we could just fix the Spectrum CSS.

Reply:
I found a way to remove the extra computed property. I just cache the `placement` attribute from the overlay content node before I run popper. I put back the old value when I tear down the popper.